### PR TITLE
[ci][test-data/3] persist ci test results

### PR DESCRIPTION
**Pull request by can-anyscale**
Persist CI test results on every test run:

Test:
- CI
- Run in postmerge, log data just fine: https://buildkite.com/ray-project/postmerge/builds/2501

### CODE REVIEW

File: ci/ray_ci/oss_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+byod:
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/ray_ci/oss_config.yaml:2]: you can use bazel runfiles to read this file, and then you do not need a copy I think.

you have already switched to the bazel workspace when reading, maybe just move this config up and out to `ci` directory or something?

or symlink in the other direction.. as `ray_release` can read symlink with no issues?

File: ci/ray_ci/BUILD.bazel [hunk 2]
```diff
@@ -32,6 +32,7 @@ py_binary(
     name = "test_in_docker",
     srcs = ["test_in_docker.py"],
     deps = [":ray_ci_lib"],
+    data = ["oss_config.yaml"],
```
Comment (aslonnie) [inline ci/ray_ci/BUILD.bazel:35]: if you are not using runfiles, you do not need this.

**Comment by can-anyscale** [comment]
Use bazel runfiles to read oss_config

File: ci/ray_ci/tester.py [hunk 3]
```diff
@@ -175,6 +177,7 @@ def main(
     if not bazel_workspace_dir:
         raise Exception("Please use `bazelisk run //ci/ray_ci`")
     os.chdir(bazel_workspace_dir)
+    init_global_config(bazel_runfile("release/ray_release/configs/oss_config.yaml"))
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:180]: it is a bit weird that this is referencing from `ray_release`, do we plan to move out this common part or something?

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/tester.py (hunk 3 above)
Comment (can-anyscale) [reply ci/ray_ci/tester.py:180]: yes, let's merge ray_release and ci so they can all share together easily

**Pull request merged**